### PR TITLE
feat: add enable wallet button

### PR DESCRIPTION
**Pull request by samh-nl**
### Details

Present an 'Enable wallet' button that allows the user to upgrade to a Gold wallet.

### Fixed Issues

$ https://github.com/Expensify/App/issues/29320
PROPOSAL: N/A, assigning was discussed via Slack [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1697034851300999?thread_ts=1697033275.944009&cid=C01GTK53T8Q)

### Tests

Preconditions: The user must not have enabled the wallet yet and must have no payment method.

1. Go to Settings
2. Go to Wallet
3. Press 'Add payment method'
4. Press 'Bank account'
5. Add the 'Plaid Checking' or 'Plaid Saving' bank account
6. Verify an 'Enable wallet' button is present in the wallet page
7. Press the 'Enable wallet' button
8. Verify the user is navigated to a page requesting additional details to be provided
9. Successfully complete the steps
10. Verify you are navigated back to the 'Wallet' page
11. Verify the wallet page does not show an 'Enable wallet' button, but instead a link stating 'Transfer balance'

- [X] Verify that no errors appear in the JS console

### Offline tests

Preconditions: The user must not have enabled the wallet yet **but must have added a payment method**.

1. Go to Settings
2. Go to Wallet
3. Verify an 'Enable wallet' button is present in the wallet page
4. Verify the button is disabled

### QA Steps

Preconditions: The user must not have enabled the wallet yet and must have no payment method.

1. Go to Settings
2. Go to Wallet
3. Press 'Add payment method'
4. Press 'Bank account'
5. Add the 'Plaid Checking' or 'Plaid Saving' bank account
6. Verify an 'Enable wallet' button is present in the wallet page
7. Press the 'Enable wallet' button
8. Verify the user is navigated to a page requesting additional details to be provided

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/137707942/2fb29f42-80f7-4edb-9b3b-f34c181d25df

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/137707942/3ef76874-5be8-4169-8212-1bf1bde84fb2

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/137707942/b76fa094-4140-4c80-b47c-7298521d00a3

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/137707942/d35606e7-4a44-4dde-bdc2-5296c81ad619

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/137707942/68c48f1a-4ef1-4704-91c6-a030977701e1

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/137707942/0909271e-6a5e-45fe-90f5-a02bb18fedbb

</details>

### CODE REVIEW

**Comment by MariaHCD** [comment]
> Regarding testing steps, I cannot get past the enable payments page to test the complete flow. Upon clicking 'Save & continue', the UpdatePersonalDetailsForWallet endpoint returns this error (no error is made visible in UI, though), please see below.

We've got an internal fix for this: https://github.com/Expensify/Web-Expensify/pull/39205

> The KYC wall currently checks if there is either a bank account or a debit card. Are both valid options for allowing upgrading the wallet or should there specifically be a bank account?

Let's restrict this to just bank accounts, debit cards are not allowed for now.

> The description on the enable payments page indicates that a payment will be processed. However, the intention here seems to only be to upgrade to a gold wallet and not to initiate any payments, or am I wrong?

Could you clarify which messaging? There will be no payments initiated (automatically) but the user can now start initiating payments. Could we also change the default text in the [ContinueButton](https://github.com/Expensify/App/blob/e6549d8cae87959bc45f8bbbcc27fce9913b95e3/src/pages/EnablePayments/ActivateStep.js#L35) to `common.continue`

**Comment by robertKozik** [comment]
@samh-nl I assume I should wait for the fix and your adjustment before C+ review?

**Comment by MariaHCD** [comment]
> If it needs to be changed for this flow, please provide the updated text.

Nice catch. Going to add the `Waiting for copy` label on the issue along with context.

**Comment by MariaHCD** [comment]
> This causes the Expensify Wallet section to not immediately appear.

Nice catch, I think this can be outside the scope of your PR. We are looking to redirect the user to the KYC flow after adding a PBA instead of back to the Wallet page: https://github.com/Expensify/App/issues/29726

**Comment by MariaHCD** [comment]
@robertKozik feel free to start on the review for this one 🙏🏼

**Comment by getusha** [comment]
Heads-up, i added `setPersonalBankAccountContinueKYCOnSuccess` but with additional param. we will just need to pass the fallback route, which is `ROUTES.SETTINGS_WALLET` here.

File: src/components/KYCWall/kycWallPropTypes.js [hunk 1]
```diff
@@ -26,6 +27,12 @@ const propTypes = {
     /** The user's wallet */
     userWallet: userWalletPropTypes,
 
+    /** Information related to the wallet activation flow */
```
Comment (MariaHCD) [inline src/components/KYCWall/kycWallPropTypes.js:30]:
```suggestion
    /** Information related to the last step of the wallet activation flow */
```

File: src/pages/settings/Wallet/WalletPage/WalletPage.js [hunk 2]
```diff
@@ -352,23 +358,44 @@ function WalletPage({bankAccountList, betas, cardList, fundList, isLoadingPaymen
...
+                                                        return;
+                                                    }
+                                                    // To allow upgrading to a gold wallet, continue with the KYC flow after adding a bank account
+                                                    BankAccounts.setPersonalBankAccountContinueKYCOnSuccess(true);
+                                                }}
                                                 enablePaymentsRoute={ROUTES.SETTINGS_ENABLE_PAYMENTS}
                                                 addBankAccountRoute={ROUTES.SETTINGS_ADD_BANK_ACCOUNT}
                                                 addDebitCardRoute={ROUTES.SETTINGS_ADD_DEBIT_CARD}
                                                 popoverPlacement="bottom"
+                                                source={hasActivatedWallet ? CONST.KYC_WALL_SOURCE.TRANSFER_BALANCE : CONST.KYC_WALL_SOURCE.ENABLE_WALLET}
```
Comment (MariaHCD) [inline src/pages/settings/Wallet/WalletPage/WalletPage.js:373]: If the wallet is already activated, we won't trigger the KYCWall at all, right? Would we need to set the source here still?

File: src/CONST.ts [hunk 3]
```diff
@@ -1062,6 +1062,12 @@ const CONST = {
         },
     },
 
+    KYC_WALL_SOURCE: {
+        REPORT: 'REPORT',
+        ENABLE_WALLET: 'ENABLE_WALLET',
+        TRANSFER_BALANCE: 'TRANSFER_BALANCE',
```
Comment (MariaHCD) [inline src/CONST.ts:1068]:
```suggestion
        REPORT: 'REPORT', // The user attempted to pay a money request
        ENABLE_WALLET: 'ENABLE_WALLET', // The user clicked on the `Enable wallet` button on the Wallet page
        TRANSFER_BALANCE: 'TRANSFER_BALANCE', // The user attempted to transfer their wallet balance to their bank account or debit card
```

Comment (MariaHCD) [reply src/CONST.ts:1068]: Just to describe each trigger

**Review by MariaHCD — COMMENTED** [review]
This looks great so far!

@samh-nl have you run into any issues when testing this? Also I think this is good for @robertKozik to review?

**Comment by MariaHCD** [comment]
> This caused a situation where I seemingly had no PBA but clicking 'Enable wallet' would still take me to the enable payments page.

Thanks for raising, I believe this PR will fix this https://github.com/Expensify/App/pull/29915/?

We're going to be looking for PBAs in the OPEN state.

**Comment by MariaHCD** [comment]
@samh-nl @allroundexperts When at the 'Additional details' wallet activation step, use the following details:

- first name: Alberta
- last name: Charleson
- dob: 08-08-2000
- ssn: 3333
- any other info for the rest of the fields

File: src/pages/settings/Wallet/WalletPage/WalletPage.js (hunk 2 above)
Comment (MariaHCD) [reply src/pages/settings/Wallet/WalletPage/WalletPage.js:373]: > I'm not sure if the wallet is deactivated automatically in that case

With the new change to [hasExpensifyPaymentMethod](https://github.com/Expensify/App/pull/29915), if the bank account is removed, hasExpensifyPaymentMethod will return false so we should be good here.

Thanks for clarifying, I think this logic makes sense 👍🏼

**Review by MariaHCD — COMMENTED** [review]
LGTM, let's test this out. Let me know if you run into any more issues when testing.

**Comment by MariaHCD** [comment]
> I've checked but `pendingAction` is a distinct state and we don't touch the value of `accountData.state` (i.e. it remains OPEN), so with the mentioned PR the PBA would still be considered valid:

Ah, interesting. To clarify, `pendingAction` will be set to `delete` if the user attempted to delete their bank account while offline. So if they attempted to enable their wallet, we'd direct them to the KYC flow even though they have a bank account that's about to be deleted. I think this should be fine because we do have backend checks that'll prevent a user from proceeding with KYC if they don't have a PBA. 

However, it's probably a simple fix to have hasExpensifyPaymentMethod also check for `bankAccount.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE`?

**Comment by getusha** [comment]
>Issue have been logged with QA team for Save and Continue button is not responding https://github.com/Expensify/Web-Expensify/pull/39174#issuecomment-1765391356

I saw this, i think this is a known issue @MariaHCD

**Comment by MariaHCD** [comment]
Looking at the BE logs to see why you both are getting blocked at the additional details step.

**Comment by MariaHCD** [comment]
@samh-nl @allroundexperts The fix has been deployed to staging, please let me know if you're still getting blocked at the additional details step 🙏🏼

**Comment by MariaHCD** [comment]
I realized you both won't be able to test steps 9 - 11 because we do not allow test activated wallets on staging. I will test on dev. @allroundexperts feel free to do the final code review, I'll do the reviewer checklist.

**Comment by MariaHCD** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Review by allroundexperts — APPROVED** [review]
Code wise, this looks good!

**Review by MariaHCD — APPROVED** [review]

**Comment by MariaHCD** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Comment by MariaHCD** [comment]
Really don't know why the reviewer checklist check is failing when I clearly have it [here](https://github.com/Expensify/App/pull/29573#issuecomment-1771443169). I've retried it multiple times.

```
Found the reviewer checklist!
You completed 0 out of 54 checklist items with 54 unfinished items
Make sure you are using the most up to date checklist found here: https://raw.githubusercontent.com/Expensify/App/main/contributingGuides/REVIEWER_CHECKLIST.md
Error: PR Reviewer Checklist is not completely filled out. Please check every box to verify you've thought about the item.
```

Going to merge anyway...

**Comment by kevinksullivan** [comment]
been trying to test this [here](https://expensify.slack.com/archives/C07J32337/p1697761413725979) but have run into unrelated issues.

**Pull request merged**